checker/tests/nullness/DefaultingForEach.java

```diff
@@ -9,6 +9,8 @@
 class DefaultForEach {
     @DefaultQualifier(Nullable.class)
     Object @NonNull [] foo() {
+        // :: error: (nullness.on.new.array)
```

Comment:
Ah, that seems less ideal. We want to say that object creations are intrinsically non-null, so even in a default nullable context, they should be non-null.
Can you try re-instating code in the ATF to achieve this? Try using `addMissingAnnotations` instead of `replaceAnnotation`, maybe that does what we want. Alternatively, we need to change the order in which defaulting works.